riscv-page-tables/src/page_table.rs

```diff
@@ -797,6 +879,50 @@ impl<T: PagingMode> PlatformPageTable<T> {
 
         Ok(pages)
     }
+
+    /// Returns a list of shareable pages for the given range. Gurantees that the full range of pages
+    /// are eligble are shareable
+    pub fn get_shareable_range<P: ShareablePhysPage>(
+        &self,
+        addr: PageAddr<T::MappedAddressSpace>,
+        page_size: PageSize,
+        num_pages: u64,
+    ) -> Result<LockedSharedPageList<P>> {
+        if page_size.is_huge() {
+            return Err(Error::PageSizeNotSupported(page_size));
+        }
+
+        let mut inner = self.inner.lock();
+
+        // First make sure the entire range can be unmapped before we start invalidating things.
```

Comment:
I don't think the first pass is necessary here since locking the page for sharing doesn't have any host-visible side-effects like invalidation does and `LockedPageList` will clean up any locked pages if we bail out early.